docs/apache-airflow/security/security_model.rst

```diff
@@ -0,0 +1,129 @@
+   code in DAG files is executed on workers. Therefore, DAG authors can create
+   and change code executed on workers and potentially access the credentials
+   that DAG code uses to access external systems. DAG Authors have full access
+   to the metadata database and internal audit logs.
+
+3. **Authenticated UI users**: They have access to the UI and API. Admin
+   users can manage permissions and execute code on workers. Connection
+   configuration users can configure connections and execute code on
+   workers. Operations users have access to DAG execution status. Trust
+   is crucial to prevent abuse and Denial of Service attacks.
+
+4. **Non-authenticated UI users**: Airflow doesn't support
+   unauthenticated users by default. If allowed, vulnerabilities must be
+   addressed by the Deployment Manager.
+
+Capabilities of authenticated UI users
+--------------------------------------
+
+The capabilities of **Authenticated UI users** can vary depending on
+what roles have been configured by the Deployment Manager or Admin users as well as what permissions those roles have. Permissions on roles can be scoped as tightly as a single DAG, for example, or as broad as Admin. Below are three general categories to help conceptualize some of the capabilities authenticated users may have:
+
+1. **Admin users**: They manage and grant permissions to other users,
+   with full access to all UI capabilities. They can potentially execute
+   code on workers by configuring connections and need to be trusted not
+   to abuse these privileges. They have access to sensitive credentials
+   and can modify them. By default, they don't have access to
+   system-level configuration. They should be trusted not to misuse
+   sensitive information accessible through connection configuration.
+   They also have the ability to create a Webserver Denial of Service
+   situation and should be trusted not to misuse this capability.
+
+2. **Connection configuration users**: They configure connections and
+   potentially execute code on workers during DAG execution. Trust is
+   required to prevent misuse of these privileges. They have full access
+   to sensitive credentials stored in connections and can modify them.
+   Access to sensitive information through connection configuration
+   should be trusted not to be abused. They also have the ability to
+   create a Webserver Denial of Service situation and should be trusted
+   not to misuse this capability.
+
+3. **Operations users**: They have access to DAG execution status via
+   the UI. Currently, Airflow lacks full protection for accessing groups
+   of DAGs' history and execution. They can perform actions such as
+   clearing, re-running, triggering DAGs, and changing parameters.
+   Depending on access restrictions, they may also have access to
+   editing variables and viewing Airflow configuration. They should not
+   have access to sensitive system-level information or connections, and
```

Comment:
Doesn't the operator role give users connection edit though? I'd maybe set rid of this whole sentence.

(I realize this isn't the intent of this PR, but reading through it with fresh eyes anyways)

Reply:
Not necessarily - You need to have Admin -> Connections access to edit connection. You can still re-run task and do all kinds of clearing etc. without it

Reply:
The role can [edit connections](https://github.com/apache/airflow/blob/2ab78ec441a748ae4d99e429fe336b80a601d7b1/airflow/www/security.py#L136).

The connections part of this should definitely move to the prior sentence of things they can access.

Reply:
I'll start a separate PR for my suggestions here 👍

Reply:
(#33021)